### PR TITLE
Legend redesign

### PR DESCRIPTION
**Pull request by JoaquinTrinanes**
### General description

Redesigns the legend according to the new design:
- Group by category
- Settings (show by category, allow selecting unit) are different than the legends themselves

Also upgraded to `react-query` v4, now `@tanstack/react-query` 🎉 

### Designs

https://www.figma.com/file/WxYMmJrUZAgV6f0Csmt8fK/LG-Tool---Work-in-progress?node-id=3182%3A68858

### Testing instructions

_Provide minimal instructions on how to test this PR._

- Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging

- [ ] Branch name / PR includes the related Jira ticket Id.
- [ ] Tests to check core implementation / bug fix added.
- [ ] All checks in Continuous Integration workflow pass.
- [ ] Feature functionally tested by reviewer(s).
- [ ] Code reviewed by reviewer(s).
- [ ] Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
-    "lint": "next lint",
+    "lint": "next lint --report-unused-disable-directives warn",
```
Comment (davidsingal) [inline client/package.json:9]: Just have a question. Why do we need this flag?

File: client/src/components/accordion/component.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import type { HTMLAttributes } from 'react';
+import { useCallback, useEffect, useState, useMemo } from 'react';
+
+type AccordionEntryProps = React.PropsWithChildren<{
+  header: React.ReactNode;
+  expanded?: boolean;
+  onExpandedChange?: (expanded: boolean) => void;
+}>;
+
+type AccordionProps = Omit<HTMLAttributes<HTMLDivElement>, 'children'> &
```
Comment (davidsingal) [inline client/src/components/accordion/component.tsx:12]: you have https://headlessui.com/react/disclosure available in HeadlessUi, did you check it? in that case, why did you decide to make a custom one?

File: client/src/hooks/fuse/index.ts [hunk 3]
```diff
@@ -14,6 +15,14 @@ const useFuse: <T>(data: T[], options?: Fuse.IFuseOptions<T>) => UseFuseReturn<T
 ) => {
   const [term, setTerm] = useState('');
 
+  const search = useCallback<UseFuseReturn<unknown>['search']>((search) => {
+    if (typeof search === 'string') {
+      setTerm(search);
+    } else {
+      setTerm(search.target.value);
+    }
+  }, []);
```
Comment (davidsingal) [inline client/src/hooks/fuse/index.ts:24]: I think this part could be improved, you are assuming things. Let's leave it as it is but add a comment to improve in the future.

File: client/package.json (hunk 1 above)
Comment (JoaquinTrinanes) [reply client/package.json:9]: We don't need it per se, but I think it's nice to warn when some directives disable warnings/errors that don't exist anymore. The same way we warn with unused variables and such

File: client/tailwind.config.js [hunk 4]
```diff
@@ -12,9 +12,9 @@ module.exports = {
         sans: ['Public Sans', 'sans-serif'],
       },
       fontSize: {
-        xxs: '10px',
-        xs: '12px',
-        '2xs': '0.625rem',
+        xs: '0.75rem', // 12px
+        '2xs': '0.625rem', // 10px
+        '3xs': '0.5rem', // 8px
```
Comment (davidsingal) [inline client/tailwind.config.js:17]: 8px?? is this in the ui-kit??

File: client/src/components/accordion/component.tsx (hunk 2 above)
Comment (JoaquinTrinanes) [reply client/src/components/accordion/component.tsx:12]: I didn't know headless UI had one, I can probably quickly swap it

**Review by davidsingal — APPROVED** [review]
I leave some questions in the code, but it works fine.

**Pull request merged**